* Unable to fetch repository from GitLab. Checksum mismatch
@ 2022-07-05 15:51 rashmi pisal
  2022-07-05 15:56 ` [yocto] " Quentin Schulz
  0 siblings, 1 reply; 2+ messages in thread
From: rashmi pisal @ 2022-07-05 15:51 UTC (permalink / raw)
  To: yocto

[-- Attachment #1: Type: text/plain, Size: 3767 bytes --]

Hello,

I am building a software image and it fails when fetching a repository form GitLab. https://github.com/ajstarks/openvg

Initially I was using git:// protocol to fetch the files but due to the GItLab ended support of git:// I am using https:// protocol. I am getting following error. Any help is appreciated. Thank you.

==================================================================
NOTE: Running task 2097 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-openembedded/meta-oe/recipes-graphics/gphoto2/libgphoto2_2.5.8.bb:do_patch)
NOTE: recipe libgphoto2-2.5.8-r0: task do_patch: Started
NOTE: recipe libgphoto2-2.5.8-r0: task do_patch: Succeeded
NOTE: Running task 2098 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-openembedded/meta-oe/recipes-support/glog/glog_0.3.4.bb:do_patch)
NOTE: recipe glog-0.3.4-r0: task do_patch: Started
NOTE: recipe glog-0.3.4-r0: task do_patch: Succeeded
NOTE: Running task 2099 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta/recipes-devtools/perl/perl_5.24.1.bb:do_packagedata)
NOTE: recipe perl-5.24.1-r0: task do_packagedata: Started
NOTE: recipe perl-5.24.1-r0: task do_packagedata: Succeeded
NOTE: Running task 2100 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-rebound/recipes-bsp/openvg/openvg.bb:do_fetch)
NOTE: recipe openvg-1.0-r0: task do_fetch: Started
WARNING: openvg-1.0-r0 do_fetch: Checksum mismatch for local file /var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg
Cleaning and trying again.
WARNING: openvg-1.0-r0 do_fetch: Renaming /var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg to /var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg_bad-checksum_ddfeecef70433040e18ee9e68481e7f3
WARNING: openvg-1.0-r0 do_fetch: Checksum failure encountered with download of https://github.com/ajstarks/openvg - will attempt other sources if available
ERROR: openvg-1.0-r0 do_fetch: Fetcher failure for URL: 'https://github.com/ajstarks/openvg'. Checksum mismatch!
File: '/var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg' has md5 checksum ddfeecef70433040e18ee9e68481e7f3 when a8b34f242afbf3cac74fc7d34a496b24 was expected
If this change is expected (e.g. you have upgraded to a new version without updating the checksums) then you can use these lines within the recipe:
SRC_URI[md5sum] = "ddfeecef70433040e18ee9e68481e7f3"
SRC_URI[sha256sum] = "7c67772c69b23d5b4cd298c7ddbf02938c3d11de04747c3243435e1463169669"
Otherwise you should retry the download and/or check with upstream to determine if the file has become corrupted or otherwise unexpectedly modified.

ERROR: openvg-1.0-r0 do_fetch: Fetcher failure for URL: 'https://github.com/ajstarks/openvg'. Unable to fetch URL from any source.
ERROR: openvg-1.0-r0 do_fetch: Function failed: base_do_fetch
ERROR: Logfile of failure stored in: /var/lib/jenkins/workspace/yocto-rebound/build/tmp/work/cortexa7hf-neon-vfpv4-poky-linux-gnueabi/openvg/1.0-r0/temp/log.do_fetch.30930
NOTE: recipe openvg-1.0-r0: task do_fetch: Failed
ERROR: Task (/var/lib/jenkins/workspace/yocto-rebound/meta-rebound/recipes-bsp/openvg/openvg.bb:do_fetch) failed with exit code '1'
NOTE: Running task 2101 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-gplv2/recipes-extended/gawk/gawk_3.1.5.bb:do_prepare_recipe_sysroot)
NOTE: recipe gawk-3.1.5-r2: task do_prepare_recipe_sysroot: Started
NOTE: recipe gawk-3.1.5-r2: task do_prepare_recipe_sysroot: Succeeded
NOTE: recipe opencv-3.3+gitAUTOINC+87c27a074d_2a9d1b22ed_a62e20676a_34e4206aef_fccf7cd6a4-r0: task do_fetch: Succeeded
NOTE: Tasks Summary: Attempted 2101 tasks of which 0 didn't need to be rerun and 1 failed.
====================================================================================

[-- Attachment #2: Type: text/html, Size: 4009 bytes --]

^ permalink raw reply	[flat|nested] 2+ messages in thread
* Re: [yocto] Unable to fetch repository from GitLab. Checksum mismatch
  2022-07-05 15:51 Unable to fetch repository from GitLab. Checksum mismatch rashmi pisal
@ 2022-07-05 15:56 ` Quentin Schulz
  0 siblings, 0 replies; 2+ messages in thread
From: Quentin Schulz @ 2022-07-05 15:56 UTC (permalink / raw)
  To: rashmipisal, yocto

Hi Rashmi,

You only need to add ;protocol=https to your SRC_URI which starts with 
git://.

git:// in SRC_URI is NOT the protocol used to download the sources, it 
is the way Bitbake knows which fetcher to use (in that case, the git 
fetcher). If you replace it with http, it's going to be the HTTP 
fetcher, which uses curl. It is not what you want to use for git repos.

c.f. 
https://docs.yoctoproject.org/bitbake/bitbake-user-manual/bitbake-user-manual-fetching.html#git-fetcher-git

Hope this helps,
Cheers,
Quentin

On 7/5/22 17:51, rashmi pisal via lists.yoctoproject.org wrote:
> Hello,
> 
> I am building a software image and it fails when fetching a repository form GitLab. https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ajstarks_openvg&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=kJKsWKz_K582lp2OiBXmm-8OF2OXSN90rZSBapaIU84&e=
> 
> Initially I was using git:// protocol to fetch the files but due to the GItLab ended support of git:// I am using https:// protocol. I am getting following error. Any help is appreciated. Thank you.
> 
> ==================================================================
> NOTE: Running task 2097 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-openembedded/meta-oe/recipes-graphics/gphoto2/libgphoto2_2.5.8.bb:do_patch)
> NOTE: recipe libgphoto2-2.5.8-r0: task do_patch: Started
> NOTE: recipe libgphoto2-2.5.8-r0: task do_patch: Succeeded
> NOTE: Running task 2098 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-openembedded/meta-oe/recipes-support/glog/glog_0.3.4.bb:do_patch)
> NOTE: recipe glog-0.3.4-r0: task do_patch: Started
> NOTE: recipe glog-0.3.4-r0: task do_patch: Succeeded
> NOTE: Running task 2099 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta/recipes-devtools/perl/perl_5.24.1.bb:do_packagedata)
> NOTE: recipe perl-5.24.1-r0: task do_packagedata: Started
> NOTE: recipe perl-5.24.1-r0: task do_packagedata: Succeeded
> NOTE: Running task 2100 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-rebound/recipes-bsp/openvg/openvg.bb:do_fetch)
> NOTE: recipe openvg-1.0-r0: task do_fetch: Started
> WARNING: openvg-1.0-r0 do_fetch: Checksum mismatch for local file /var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg
> Cleaning and trying again.
> WARNING: openvg-1.0-r0 do_fetch: Renaming /var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg to /var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg_bad-checksum_ddfeecef70433040e18ee9e68481e7f3
> WARNING: openvg-1.0-r0 do_fetch: Checksum failure encountered with download of https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ajstarks_openvg&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=kJKsWKz_K582lp2OiBXmm-8OF2OXSN90rZSBapaIU84&e=  - will attempt other sources if available
> ERROR: openvg-1.0-r0 do_fetch: Fetcher failure for URL: 'https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ajstarks_openvg&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=kJKsWKz_K582lp2OiBXmm-8OF2OXSN90rZSBapaIU84&e= '. Checksum mismatch!
> File: '/var/lib/jenkins/workspace/yocto-rebound/build/downloads/openvg' has md5 checksum ddfeecef70433040e18ee9e68481e7f3 when a8b34f242afbf3cac74fc7d34a496b24 was expected
> If this change is expected (e.g. you have upgraded to a new version without updating the checksums) then you can use these lines within the recipe:
> SRC_URI[md5sum] = "ddfeecef70433040e18ee9e68481e7f3"
> SRC_URI[sha256sum] = "7c67772c69b23d5b4cd298c7ddbf02938c3d11de04747c3243435e1463169669"
> Otherwise you should retry the download and/or check with upstream to determine if the file has become corrupted or otherwise unexpectedly modified.
> 
> ERROR: openvg-1.0-r0 do_fetch: Fetcher failure for URL: 'https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ajstarks_openvg&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=kJKsWKz_K582lp2OiBXmm-8OF2OXSN90rZSBapaIU84&e= '. Unable to fetch URL from any source.
> ERROR: openvg-1.0-r0 do_fetch: Function failed: base_do_fetch
> ERROR: Logfile of failure stored in: /var/lib/jenkins/workspace/yocto-rebound/build/tmp/work/cortexa7hf-neon-vfpv4-poky-linux-gnueabi/openvg/1.0-r0/temp/log.do_fetch.30930
> NOTE: recipe openvg-1.0-r0: task do_fetch: Failed
> ERROR: Task (/var/lib/jenkins/workspace/yocto-rebound/meta-rebound/recipes-bsp/openvg/openvg.bb:do_fetch) failed with exit code '1'
> NOTE: Running task 2101 of 5013 (/var/lib/jenkins/workspace/yocto-rebound/meta-gplv2/recipes-extended/gawk/gawk_3.1.5.bb:do_prepare_recipe_sysroot)
> NOTE: recipe gawk-3.1.5-r2: task do_prepare_recipe_sysroot: Started
> NOTE: recipe gawk-3.1.5-r2: task do_prepare_recipe_sysroot: Succeeded
> NOTE: recipe opencv-3.3+gitAUTOINC+87c27a074d_2a9d1b22ed_a62e20676a_34e4206aef_fccf7cd6a4-r0: task do_fetch: Succeeded
> NOTE: Tasks Summary: Attempted 2101 tasks of which 0 didn't need to be rerun and 1 failed.
> ====================================================================================
> 
> 
> 
> -=-=-=-=-=-=-=-=-=-=-=-
> Links: You receive all messages sent to this group.
> View/Reply Online (#57456): https://urldefense.proofpoint.com/v2/url?u=https-3A__lists.yoctoproject.org_g_yocto_message_57456&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=aiy2liQy8V0E54f0u0J9akaQ-NaVMPXg8DU7h-wYdME&e=
> Mute This Topic: https://urldefense.proofpoint.com/v2/url?u=https-3A__lists.yoctoproject.org_mt_92187673_6293953&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=ss4rCycim4dMwzxGO8iJtHaJO4Xp48AcfJR1y8wCLes&e=
> Group Owner: yocto+owner@lists.yoctoproject.org
> Unsubscribe: https://urldefense.proofpoint.com/v2/url?u=https-3A__lists.yoctoproject.org_g_yocto_unsub&d=DwIFaQ&c=_sEr5x9kUWhuk4_nFwjJtA&r=LYjLexDn7rXIzVmkNPvw5ymA1XTSqHGq8yBP6m6qZZ4njZguQhZhkI_-172IIy1t&m=Uy8eY6eBlqV_AsuEY-zJDusZsvsTvDx2FOD4Nhn0s-1R6JDqU0aLQCXt1Sje117Y&s=8zuU9VVW0HPYB502lx6SL7lcHAkKaLbUSgkqz_we0tM&e=  [quentin.schulz@theobroma-systems.com]
> -=-=-=-=-=-=-=-=-=-=-=-
> 


^ permalink raw reply	[flat|nested] 2+ messages in thread
end of thread, other threads:[~2022-07-05 15:56 UTC | newest]

Thread overview: 2+ messages (download: mbox.gz / follow: Atom feed)
-- links below jump to the message on this page --
2022-07-05 15:51 Unable to fetch repository from GitLab. Checksum mismatch rashmi pisal
2022-07-05 15:56 ` [yocto] " Quentin Schulz

